Plugin/CategoryFieldPlugin.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace AvS\ScopeHint\Plugin;
+
+use Magento\Catalog\Api\CategoryRepositoryInterface;
+use Magento\Framework\Escaper;
+use Magento\Framework\Registry;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Ui\Component\Form\Field as FieldComponent;
+
+class CategoryFieldPlugin
+{
+    /**
+     * @var StoreManagerInterface
+     */
+    protected $storeManager;
+
+    /**
+     * @var Registry
+     */
+    protected $registry;
+
+    /**
+     * @var CategoryRepositoryInterface
+     */
+    protected $categoryRepository;
+
+    /**
+     * @var Escaper
+     */
+    protected $escaper;
+
+    /**
+     * @param StoreManagerInterface $storeManager
+     * @param Registry $registry
+     * @param CategoryRepositoryInterface $categoryRepository
+     * @param Escaper $escaper
+     */
+    public function __construct(
+        StoreManagerInterface $storeManager,
+        Registry $registry,
+        CategoryRepositoryInterface $categoryRepository,
+        Escaper $escaper
+    ) {
+        $this->storeManager = $storeManager;
+        $this->registry = $registry;
+        $this->categoryRepository = $categoryRepository;
+        $this->escaper = $escaper;
+    }
+
+    /**
+     * @param FieldComponent $subject
+     * @param $config
+     * @return array
+     */
+    public function afterGetConfiguration(FieldComponent $subject, $config)
+    {
+        if ($subject->getContext()->getNamespace() == 'category_form' &&
+            $config['formElement'] != 'hidden' &&
+            strpos($config['dataScope'], 'use_config.') === false
+        ) {
+            $scopeHints = [];
+            $attributeCode = $config['dataScope'];
+            $storeViews = $this->storeManager->getStores();
```

Comment:
In `\AvS\ScopeHint\Plugin\ProductEavDataProviderPlugin`, there is a local cache for the stores, which may be a good idea if called a lot, but not sure how often this is actually called.

Reply:
Yeah, I've seen this but after some digging you can seen that the database query is already cached natively. The local cache doesn't really do that much.